reframe/core/pipeline.py

```diff
@@ -1487,26 +1511,29 @@ def __str__(self):
                                                self.name, self.prefix)
 
 
-class RunOnlyRegressionTest(RegressionTest):
+class RunOnlyRegressionTest(RegressionTest, extended_test=True):
     '''Base class for run-only regression tests.
 
     This class is also directly available under the top-level :mod:`reframe`
     module.
     '''
 
+    @final
```

Comment:
Avoiding the discard above, only the final keyword at the base class will be needed.